src/workerd/io/actor-cache.c++

```diff
@@ -2336,12 +2336,12 @@ kj::Promise<void> ActorCache::flushImpl(uint retryCount) {
         // have to perform the delete first in order to determine the count that the delete() call
         // should return.
         CountedDeleteFlush* countedDeleteFlush;
-        KJ_IF_MAYBE(i, c->get()->flushIndex) {
-          countedDeleteFlush = &countedDeleteFlushes[*i];
+        KJ_IF_SOME(i, c->flushIndex) {
+          countedDeleteFlush = &countedDeleteFlushes[i];
         } else {
-          c->get()->flushIndex = countedDeleteFlushes.size();
+          c->flushIndex = countedDeleteFlushes.size();
           countedDeleteFlush = &countedDeleteFlushes.add(CountedDeleteFlush{
-            .countedDelete = kj::addRef(**c).attach(kj::defer([&cd = *c->get()]() mutable {
+            .countedDelete = kj::addRef(*c).attach(kj::defer([&cd = *c]() mutable {
```

Comment:
Not a bug in your PR, but this feels like a potential lifetime bug to me...  `kj::Own` destroys attachments after the owned object, so if the `addRef()`ed reference happens to be the last reference, it will be destroyed before the `kj::defer()` block attempts to write to its `flushIndex` member.  Maybe there's a condition that ensures this never happens, or there's some subtlety in the following comment that I'm missing.  But I think it would probably be safer if the `kj::defer()` block captured its own `addRef()`ed reference?

Reply:
So I can tell you that it happens to be safe _right now_ because we carry forward counted deletes when we replace a dirty entry with another and dirty entries are never evicted, only replaced with a clean when the flush completes. That said, I agree that this is not obvious whatsoever and I'm working on making this more obvious in the PR that will follow this one.